truetype/truetype.go

```diff
@@ -253,6 +268,10 @@ func (f *Font) parseCmap() error {
 			return FormatError("inconsistent cmap length")
 		}
 		offset += 16
+		// check cmap length for reading nGroups*3*4 bytes starting at offset
```

Comment:
It may be worth checking, a few lines above, that nGroups is a sane value, so that 12*nGroups doesn't overflow. For example:

nGroups := u32(etc)
if nGroups > 0x10000 {
  return UnsupportedError(etc)
}
if length != etc {